benchmarks/transport/ghex_p2p_bi_cb_avail_mt.cpp

```diff
@@ -151,7 +150,10 @@ int main(int argc, char *argv[])
                 make_zero(rmsgs[j]);
             }
 
-            comm.barrier();
+#ifdef USE_OPENMP
+#pragma omp barrier
+#endif
+            MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
@mbianco We cannot execute MPI barrier here. It has to be our barrier, which would progress the communication engine. MPI barrier does not progress our workers and leads to a dead-lock.

The previous barrier implementation from `communicator` has to be available somehow and executed here.